### PR TITLE
Implement compatible_installer

### PR DESCRIPTION
**Pull request by NeoTheThird**
for https://github.com/ubports/ubports-installer/issues/2509

### CODE REVIEW

File: v2/checkSemver.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  .then(([latest_installer, devices]) =>
+    Promise.all(
+      devices.map(device =>
+        semver.satisfies(latest_installer, device.compatible_installer, {
+          includePrerelease: true
+        })
+          ? Promise.resolve()
+          : Promise.reject(
+              new Error(
+                `${device.codename}: ${device.name}: compatible_installer ${device.compatible_installer} does not satisfy latest installer version ${latest_installer}`
```
Comment (Flohack74) [inline v2/checkSemver.js:46]: Do I understand right: This means that before an installer config can be merged at least one compatible installer version must exist in the releases section of installer repo.

File: v2/checkSemver.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+                }))
+                .filter(({ compatible_installer }) => compatible_installer)
+            )
+        )
+      )
+    )
+    .then(devices => devices.flat())
+])
+  .then(([latest_installer, devices]) =>
+    Promise.all(
```
Comment (maciek134) [inline v2/checkSemver.js:38]: Is this needed? Nothing below is actually a promise, so you could just
```
devices.forEach((device) => {
  const satisfied = semver.satisfies(latest_installer, device.compatible_installer, {
    includePrerelease: true
  });
  if (!satisfied) {
    throw new Error(...);
  }
});
```
without returning anything.

File: v2/checkSemver.js (hunk 1 above)
Comment (NeoTheThird) [reply v2/checkSemver.js:46]: no. it gets the latest release and checks if everything is compatible.

Comment (Flohack74) [reply v2/checkSemver.js:46]: Yes. What I mean is: If I create now an installer config that requires 0.9.4 it couldnt be merged until this installer is released.

Comment (NeoTheThird) [reply v2/checkSemver.js:46]: yeah. right, i should probably not do it like this :]

File: v2/checkSemver.js (hunk 2 above)
Comment (NeoTheThird) [reply v2/checkSemver.js:38]: guess so :)

File: v2/checkSemver.js (hunk 1 above)
Comment (Flohack74) [reply v2/checkSemver.js:46]: Well it depends: If we want a 100% working state its ok to require it. My thought was just to question it, not to turn it down. If we go that way, we should just know this and also tell the world that this behaviour changes.

Comment (maciek134) [reply v2/checkSemver.js:46]: I'm all for leaving this in, one less opportunity for a typo or miscommunication

Comment (NeoTheThird) [reply v2/checkSemver.js:49]: i'm leaving the script in, but won't assert it's passing for ci to greenlight a PR. that way you can still check manually if you want

**Comment by NeoTheThird** [comment]
Alright, this has gotten interesting! I finally looked into writing ajv plugins for more advanced schema validations. this now allows us to specify a `ubports_installer-compatibility` semver range in the schema of every action and to validate that the `compatible_installer` semver range specified on an `operating_system` of a device config is a subset of this. Straightforward syntax, straightforward validation.

The scaffolding of this plugin can be used to do much more validation later, very excited to see where this will go!

**Pull request merged**